.github/workflows/deploy.yml

```
@@ -0,0 +1,19 @@
+name: Deploy
+
+on:
+  push:
+  merge_group:
+  release:
+    types:
+      - published
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: [3.8, 3.9, '3.10']
```

```suggestion
        python-version: [3.8, 3.9, '3.10']
        publish: ${{ github.event_name == 'release' && github.event.action == 'published' && matrix.os == 'ubuntu-latest' && matrix.python-version == '3.9' }}
```

@AleCandido @andrea-pasquale, are we missing this, right?

---

Yes, I think so. Although maybe this line should be added below line 16 (see my comment) (?)

---

Ah, right: for me, every time this workflow is run is just to actually deploy, so I usually do not put such a key.

But if you want to build wheels on `push:` as well you definitely need to filter out this run (you do not want to deploy a version at each push).
And since it is a pure Python package, you'd be building OS independent wheels, so deploying from more than one OS would cause a clash and a workflow failure (even though the wheel would be actually deployed).